k8s-api/src/main/java/io/enmasse/k8s/api/CachingSchemaProvider.java

```diff
@@ -8,23 +8,32 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class CachingSchemaProvider implements SchemaProvider, Watcher<Schema> {
     private static final Logger log = LoggerFactory.getLogger(CachingSchemaProvider.class);
     private volatile Schema schema = null;
+    private final List<SchemaListener> listeners = new ArrayList<>();
 
     @Override
     public Schema getSchema() {
         return schema;
     }
 
+    public void registerListener(SchemaListener listener) {
+        listeners.add(listener);
+    }
+
     @Override
     public void onUpdate(List<Schema> items) {
```

Comment:
This code doesn't read well. It seems arbitrary to ignore all but the first entry in the list.

Reply:
I assume you mean the existing code. Yes, it will take the first since there is only 1 Schema. The SchemaApi/KubeSchemaApi should instead take the SchemaListener interface rather than Watcher<Schema> . I will change this at a later point to avoid too big a code change.

Reply:
Yeah, I saw it was existing code that cached _the_ schema from the list.  Happy to defer a refactor until a better time.